cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -164,24 +158,47 @@ private void resetCompatibilityOnSubject(
     }
   }
 
-  private String registerSchemaPostCall(ClusterSchemaRequest clusterSchemaRequest) {
+  private RegisterSchemaCustomResponse registerSchemaPostCall(
+      ClusterSchemaRequest clusterSchemaRequest) {
     String suffixUrl =
         clusterSchemaRequest.getEnv()
-            + "/subjects/"
+            + "/"
+            + SCHEMA_SUBJECTS_URI
+            + "/"
             + clusterSchemaRequest.getTopicName()
-            + "-value/versions";
+            + SCHEMA_VALUE_URI
+            + "/versions";
     Pair<String, RestTemplate> reqDetails =
         clusterApiUtils.getRequestDetails(suffixUrl, clusterSchemaRequest.getProtocol());
 
     HttpEntity<Map<String, String>> request =
         buildSchemaEntity(
             clusterSchemaRequest.getFullSchema(), clusterSchemaRequest.getClusterIdentification());
-    ResponseEntity<String> responseNew =
-        reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
 
-    String updateTopicReqStatus = responseNew.getBody();
-    log.info("SchemaRequest response body {}", responseNew.getBody());
-    return updateTopicReqStatus;
+    ResponseEntity<RegisterSchemaResponse> schemaResponseResponseEntity =
+        reqDetails
+            .getRight()
+            .postForEntity(reqDetails.getLeft(), request, RegisterSchemaResponse.class);
+    List<Integer> versionsListAfter =
+        new ArrayList<>(
+            Objects.requireNonNull(
+                getSchemaVersions(
+                    clusterSchemaRequest.getEnv(),
+                    clusterSchemaRequest.getTopicName(),
+                    clusterSchemaRequest.getProtocol(),
+                    clusterSchemaRequest.getClusterIdentification())));
+
+    RegisterSchemaResponse registerSchemaResponse = schemaResponseResponseEntity.getBody();
+    RegisterSchemaCustomResponse registerSchemaCustomResponse = new RegisterSchemaCustomResponse();
+    registerSchemaCustomResponse.setId(
+        registerSchemaResponse != null ? registerSchemaResponse.getId() : -1);
+
```

Comment:
if we are setting ids to -1 should we be throwing an error here is there any chance that would cause an issue?